AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift

```diff
@@ -41,3 +42,28 @@ public class AmplifyAWSServiceConfiguration: AWSServiceConfiguration {
         super.init(region: regionType, credentialsProvider: nil)
     }
 }
+
+extension AmplifyAWSServiceConfiguration {
+
+    static var platformMapping: [Platform: String] = [:]
+
+    public static func addUserAgentPlatform(_ platform: Platform, version: String) {
+        guard platform != .ios else {
+            return
+        }
+        platformMapping[platform] = version
+    }
+
+    public enum Platform: String {
+
+        case ios = "amplify-iOS"
+
+        case flutter = "amplify-flutter"
+    }
+
+    static func platformInformation() -> String {
+        var mapping = platformMapping
+        mapping[.ios] = version
+        return mapping.map {"\($0.rawValue)/\($1)"}.joined(separator: " ")
```

Comment:
Is the metrics team OK with the order of the UA components being non-deterministic?

Reply:
Will check, hopefully this wont be a problem since we have the same implementation in Android https://github.com/aws-amplify/amplify-android/pull/703/files

Reply:
Verified that the value need to be prepended.